src/yio-interface/integrationinterface.h

```diff
@@ -56,23 +56,36 @@ class IntegrationInterface {
 
     /**
      * @brief getAllAvailableEntities Can be implemented by integration
+     * @returns list of QVariantMap
+     * https://github.com/YIO-Remote/documentation/wiki/developer-API-remote#get-available-entities-json-of-loaded-entities
      */
-    virtual QStringList getAllAvailableEntities() = 0;
+    virtual QVariantList getAllAvailableEntities() = 0;
 
     /**
      * @brief sendCommand Must be implemented as Q_INVOKABLE
+     * @param entityId
      * @param type
-     * @param entity_id
+     * @param integration
+     * @param friendlyName
+     * @param supportedFeatures
+     */
+    virtual bool addAvailableEntity(const QString& entityId, const QString& type, const QString& integration,
+                                    const QString& friendlyName, const QStringList& supportedFeatures) = 0;
+
+    /**
+     * @brief sendCommand Must be implemented as Q_INVOKABLE
```

Comment:
Why does it need to be a Q_INVOKABLE?
Is there any QML dependency to the integration interface?